### ISSUE TITLE
Creating a custom patchset against upstream Wine

### ISSUE THREAD

**Issue by shmerl** [Frogging-Family/wine-tkg-git#970]
I'm trying to figure out how to create a patchset for Wine+fsync on top of current upstream Wine (let's say version 8.3).

What is the easiest way to do it using wine-tgk-git?

**Comment by csutcliff** [comment]
what's wrong with setting `_use_fastsync="true"` and building that?

**Comment by shmerl** [comment]
I just want basically an upstream Wine with only fsync patches. So I'm interested in figuring out how to get the patches against specific Wine version, similar to how Wine staging does it.

Also, I thought fastsync / winesync was kind of stalled and fsync is a preferred method until winesync will be picked up again? I haven't really followed all the developments.

**Comment by csutcliff** [comment]
yeah sorry I misread fsync as fastsync as it's what I use. Same thing applies though, enable fsync, specify a wine version and disable everything else in the config

**Comment by shmerl** [comment]
I'm not really too familiar with how to do it, so what I tried was:

* Cloned the repo: https://github.com/Frogging-Family/wine-tkg-git.git
* Copied `wine-tkg-git/wine-tkg-profiles/wine-tkg-mainline.cfg` as `wine-tkg-git/wine-tkg-profiles/wine-tkg-mainline_fsync.cfg`
* Set `_use_fsync="true"` in that new `wine-tkg-mainline_fsync.cfg`
* Then set `_LOCAL_PRESET="mainline_fsync"` in `wine-tkg-git/customization.cfg` and tried running `non-makepkg-build.sh`.
It created modified Wine sources in /tmp/...

But doesn't look like it included fsync patches there. I guess I'm doing something wrong. So do you have a suggestion how to do it differently?

**Comment by Tk-Glitch** [comment]
For mainline you might need to set `_use_esync="true"` as it's a dependency for fsync.

**Comment by shmerl** [comment]
Btw, do you recommend using fsync vs winesync or they are comparable in practice?

**Comment by shmerl** [comment]
Btw, thanks for the hint! Looks like enabling both esync and fsync flags allowed it to create a source with fsync patches. I'll check if that works.

**Comment by shmerl** [comment]
Btw, I made a diff with the patches: https://pastebin.com/raw/ijZkZdDg

I noticed it has something unrelated to fsync, like some OpenCL stuff? Looks like even in mainline preset, it applies something more, unless I'm missing something.

**Comment by shmerl** [comment]
OK, I managed to make the script itself produce patch files by adding:

```
_generate_patchsets="true"
```

to `advanced-customization.cfg`, so now it splits hotfixes into a separate file.

But then I still see that it added a few patches I didn't expect (they also show up in the script log):

```
 -> Applying steam.patch
 -> Applying fsync-unix-mainline.patch
 -> Applying fsync_futex_waitv.patch
 -> Applying proton_battleye.patch
 -> Applying proton-eac_bridge.patch
```

I guess I need to figure out how to disable `steam.patch`, `proton_battleye.patch` and `proton-eac_bridge.patch`.

**Comment by shmerl** [comment]
Managed to turn it off with

```
_proton_eac_support="false"
_proton_battleye_support="false"
```

Looking how to disable steam patch and hotfixes now.

**Comment by shmerl** [comment]
Looks like it ignores `_steam_fix="false"` directive in that case for whatever reason.

**Comment by shmerl** [comment]
So close, but not quite. A few things to clarify:

1. How to disable all hotfixes?
2. How to turn off steam patches?
3. And it's probably good to add the above and these to mainline profile somehow:
```
_proton_eac_support="false"
_proton_battleye_support="false"
```
